### PR TITLE
[ROCM] enable lu_op (#1433)

### PR DESCRIPTION
**Pull request by micmelesse**
This prs the lu_op on rocm.

### CODE REVIEW

**Comment by micmelesse** [comment]
@bloops @gbaned Is there anything I should change in this PR?

**Comment by stevenireeves** [comment]
@cheshire @gbaned can we get an update on this PR? Been open for 22 days.

**Comment by stevenireeves** [comment]
@cheshire @gbaned can we get an update on the status of this PR?

File: tensorflow/core/kernels/linalg/lu_op_gpu.cu.cc [hunk 1]
```diff
@@ -182,9 +182,16 @@ class LuOpGpu : public AsyncOpKernel {
       auto packed_triangular_factors_ptrs = solver->GetScratchSpace<uint8>(
           sizeof(Scalar*) * batch_size, "packed_triangular_factors_ptrs",
           /* on_host */ true);
+#if GOOGLE_CUDA
```
Comment (cheshire) [inline tensorflow/core/kernels/linalg/lu_op_gpu.cu.cc:185]: Does the code below need to be duplicated?

Comment (micmelesse) [reply tensorflow/core/kernels/linalg/lu_op_gpu.cu.cc]: @cheshire removed duplicated code.

**Comment by micmelesse** [comment]
@cheshire @gbaned Issues have been addressed.

**Comment by gbaned** [comment]
@cheshire Can you please review this PR ? Thanks!

**Comment by micmelesse** [comment]
@cheshire @gbaned Is there anything we can do?

**Comment by gbaned** [comment]
@cheshire Can you please review this PR ? Thanks!

**Comment by micmelesse** [comment]
@cheshire Any updates on this

**Review by cheshire — APPROVED** [review]

**Comment by micmelesse** [comment]
@cheshire What should we do about the ci failures? There is no link for more details for the failures. Is it unrelated to this pr?

**Review by gbaned — APPROVED** [review]

**Pull request merged**